src/parser/ParsedQuery.h

```diff
@@ -357,16 +357,7 @@ class ParsedQuery {
     // Select All (Select '*')
     // or
     // explicit variables selection (Select 'var_1' ... 'var_n')
-    [[nodiscard]] auto& getAccordinglySelectVariables() {
-      return isAsterisk() ? _variablesFromQueryBody
-                          : std::get<std::vector<string>>(_varsOrAsterisk);
-    }
-
-    // Get the variables accordingly to established Selector:
-    // Select All (Select '*')
-    // or
-    // explicit variables selection (Select 'var_1' ... 'var_n')
-    [[nodiscard]] const auto& getAccordinglySelectVariables() const {
+    [[nodiscard]] const auto& getSelectedVariable() const {
       return isAsterisk() ? _variablesFromQueryBody
                           : std::get<std::vector<string>>(_varsOrAsterisk);
     }
```

Comment:
It should be `getSelectedVariables()` (with a plural `s`).

And to make this interface even less error-prone,
I think this class should have exactly the following public methods:

`setAsterisk()` (or better called `setAllVariablesSelected`)
`setManuallySelected(std::vector<string>)` (register all the selected variables at once, you also read them at once in the parser.
`registerVariableVisibleInQueryBody` (exactly as it is now)
` getSelectedVariables` (as now, return the correct variables, no matter if they were manually selected or via `*`

Reply:
With this interface, it is then impossible to trigger the `std::get, wrong index` exception.